### PR TITLE
Add verify_cert option

### LINKED ISSUE #12: Option for disabling ssl verification
We should support disabling ssl cert validation, moved from: https://github.com/logstash-plugins/logstash-input-http_poller/issues/32

### PR DESCRIPTION
**Pull request by andrewvc**
Fixes #12

### CODE REVIEW

File: lib/logstash/plugin_mixins/http_client.rb [hunk 1]
```diff
@@ -39,7 +39,11 @@ def setup_http_client_config
     # How many times should the client retry a failing URL? Default is `0`
     config :automatic_retries, :validate => :number, :default => 0
 
-    # If you need to use a custom X.509 CA (`.pem` certs) specify the path to that here
+    # Set this to false to disable SSL/TLS certificate validation
+    # Note: setting this to false is generally considered insecure!
+    config :verify_cert, :validate => :boolean, :default => true
```
Comment (suyograo) [inline lib/logstash/plugin_mixins/http_client.rb]: Thoughts about naming this `:ssl_certificate_verification`. Just so we can be consistent with `es-output` https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/master/lib/logstash/outputs/elasticsearch.rb#L282

**Comment by suyograo** [comment]
Is there a way we could warn (to console) the user when this option is used? May be in `register` ?

File: lib/logstash/plugin_mixins/http_client.rb (hunk 1 above)
Comment (andrewvc) [inline lib/logstash/plugin_mixins/http_client.rb]: @suyograo makes sense, will do (though I'll point out that SSL has been replaced by TLS at this point).

**Comment by andrewvc** [comment]
@suyograo I don't have access to `#register` from this since its a mixin. I personally think its fine not to warn since it is something the user has to explicitly set. What's the use case you're envisioning here?

**Comment by suyograo** [comment]
@andrewvc i'm cool with that :+1:

**Comment by andrewvc** [comment]
@suyograo I just fixed the issue with inconsistent parameter names. Looking better?

**Comment by suyograo** [comment]
LGTM

**Comment by andrewvc** [comment]
Thanks @suyograo !

**Pull request closed without merging**